File: stacks_queues/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
    stack<vector<int>> st;
    vector<int> sol(temperatures.size(),0);
    for(int iii=0;iii<temperatures.size();iii++)
    {
    while(!st.empty() && temperatures[iii] > st.top()[0]){
    int id {st.top()[1]};
    sol[id] = iii-id;
    st.pop();
    }
    st.push(vector<int>{temperatures[iii],iii});
    }
    return sol;
    }
};

File: stacks_queues/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
    int[] resultDays = new int[temperatures.length];
    Stack<Integer> temp = new Stack<Integer>();

    for(int i = 0 ; i < temperatures.length ; i ++ )
    {
        while(!temp.isEmpty() && temperatures[i] >  temperatures[temp.peek()]){
        int index = temp.pop();
        resultDays[index] = i - index;
        }
        temp.push(i);
    }   

    return resultDays;
    }
}
